### PR TITLE
strutils.toHex: accept openarray[byte]

### PR DESCRIPTION
**Pull request by ghost**
This fixes a mild annoyance when `toHex` doesn't accept `seq[byte]`.

### CODE REVIEW

**Comment by Araq** [comment]
This is a breaking change and a major annoyance is to use `openArray[char]` everywhere in strutils, just in case. `parseutils` offers things for openArray of char, so add it there if it doesn't exist yet.

**Comment by ghost** [comment]
How about another proc in `strutils` for bytes? Hexifying integers is already in there.

File: lib/pure/strutils.nim [hunk 1]
```diff
@@ -1036,14 +1038,32 @@ func toHex*(s: string): string {.rtl.} =
...
 
-  const HexChars = "0123456789ABCDEF"
   result = newString(s.len * 2)
   for pos, c in s:
     var n = ord(c)
     result[pos * 2 + 1] = HexChars[n and 0xF]
     n = n shr 4
     result[pos * 2] = HexChars[n]
 
+func toHex*(data: openarray[byte]): string {.rtl.} =
```
Comment (Araq) [inline lib/pure/strutils.nim:1048]: As I said, it belongs to parseutils and should probably store the result into a `var openArray[char]`. If you avoid strings, avoid them properly.

**Comment by ghost** [comment]
There is nothing in parseutils for converting to hex.

**Pull request closed without merging**